nemo/collections/asr/parts/perturb.py

```diff
@@ -52,6 +57,23 @@
     HAVE_NUMBA = False
 
 
+def read_one_audiosegment(manifest, target_sr, rng, tarred_audio=False, audio_dataset=None):
+    if tarred_audio:
+        audio_file, file_id = next(audio_dataset)
```

Comment:
Check if audio dataset is none or not before next() and raise proper error instead of null exceptions.